src/phantasm-renderer/reflection/resources.hh

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <type_traits>
+
+#include <reflector/introspect.hh>
+
+#include <phantasm-hardware-interface/arguments.hh>
+
+#include <phantasm-renderer/resources.hh>
+
+namespace pr
+{
+namespace detail
+{
+struct arg_shape_visitor
+{
+    phi::arg::shader_arg_shape shape;
+
+    template <class T>
+    void operator()(T const& ref, char const* name)
+    {
+        // TODO: do checks on T, decide of SRV, UAV, Sampler or part of merged CBV

```

Comment:
How does this look like? In my mind it's:
1. Explicit overload for `T` = `handle::resource`, which makes it a SRV, or UAV if "write" tag is present (view is implicit, somehow should be specifiable)
2. Everything that is just plain data is merged into an implicitly created upload buffer, which constitutes a CBV. This would have to be uploaded / flushed at record time.
3. If `OuterT` derives `pr::immediate`, `sizeof(OuterT)` must be <= 8 bytes, and no `handle::resource`s are accepted.
4. Samplers have to fit into this somehow as well